### DISCUSSION TITLE
Wrong destination latitude in PGN 129284 (Seatalk1 -> NG)

### DISCUSSION THREAD

**Discussion by mglonnro** [canboat/canboat#296] [Q&A] [answered]
I have an old Raymarine E120 which is sending out waypoint information using Seatalk1, and somewhere along the way some significant information is dropped away from the latitude (but not the longitude). Any ideas where the problem might be?

1) On the Raymarine E120, GOTO cursor / waypoint. 

2) Seatalk1 data is passed through Raymarine's Seatalk -> SeatalkNG converter. 

3) Yacht Devices USB-GW connected to the SeatalkNG network. 

4) `actisense-serial` gets:

`2022-09-10T06:45:24.454Z,3,129284,115,255,34,08,60,21,01,00,01,ff,ff,ff,ff,ff,ff,ff,ff,c7,98,ff,ff,ff,ff,01,00,00,00,99,1e,00,00,64,24,42,0d,ff,ff
` 

Where (I think) `99,1e,00,00` is latitude and `64,24,42,0d` is longitude. 

5) `analyzer` decodes: 

`2022-09-10T06:45:24.454Z 3 115 255 129284 Navigation Data:  SID = 8; Distance to Waypoint = 740.80 m; Course/Bearing reference = Magnetic; Perpendicular Crossed = No; Arrival Circle Entered = No; Calculation Type = Great Circle; ETA Time = Unknown; ETA Date = Unknown; Bearing, Origin to Destination Waypoint = Unknown; Bearing, Position to Destination Waypoint = 224.1 deg; Origin Waypoint Number = Unknown; Destination Waypoint Number = 1; Destination Latitude =  0.0007833; Destination Longitude = 22.2438500; Waypoint Closing Velocity = -0.01 m/s`

The real "Destination Latitude" would be somewhere over 60 deg.

**Comment by keesverruijt** [comment, marked as answer]
Your analysis is correct. 99 1e 00 00 are the 4 bytes that form the destination latitude, and they contain incorrect values.

The NMEA2000 -> USB converter doesn't actually need to understand the data bytes, so it is unlikely to mess up a particular field but not others. So that leaves the Raymarine ST1 -> NG converter, which needs to convert SeaTalk 1 datagrams to NMEA2000.

I verified that `analyzer` does this correctly for me, on board my own boat.

Example:

    2022-09-10T12:10:33.590Z,3,129284,25,255,34,6b,3d,02,03,00,00,ff,ff,ff,ff,ff,ff,94,9f,94,9f,00,00,00,00,01,00,00,00,dd,10,b1,1f,43,a9,38,03,00,00

Analyses to: 

    2022-09-10T12:10:33.590Z 3  25 255 129284 Navigation Data:  SID = 107; Distance to Waypoint = 1.97181 km; Course/Bearing reference = True; Perpendicular Crossed = No; Arrival Circle Entered = No; Calculation Type = Great Circle; ETA Time = Unknown; ETA Date = Unknown; Bearing, Origin to Destination Waypoint = 234.1 deg; Bearing, Position to Destination Waypoint = 234.1 deg; Origin Waypoint Number = 0; Destination Waypoint Number = 1; Destination Latitude = 53.1697885; Destination Longitude =  5.4044995; Waypoint Closing Velocity = 0.00 m/s

Note that the waypoint closing velocity in your raw capture says `ff ff` which looks as if the sender thinks it is an unsigned field without a value. I have multiple logs of devices producing `ff 7f` in those 2 bytes, which indicates the field should be signed. Either the NMEA 2000 standard changed, or your gateway is producing the wrong value. Anyway, it explains why `analyzer` shows `Waypoint Closing Velocity = -0.01 m/s`.

**Reply by mglonnro** [reply]
Thank you for your reply 👍

So it seems likely that my eventual fix for this will be to upgrade the old chart plotter and the other Seatalk1 stuff (including the converter). 

If I come by the actual sentence the E120 is sending, I'll try to confirm that the ST1=>NG converter is doing this. It's the standard "Made in China" Raymarine five-port converter, however, so a bit strange.
